### PR TITLE
added geneve proxy xgress

### PR DESCRIPTION
**Pull request by dariuszSki**

### CODE REVIEW

File: router/xgress_geneve/listener.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+)
+
+type listener struct{}
+
+func (self *listener) Listen(string, xgress.BindHandler) error {
+	go func() {
+		log := pfxlog.Logger()
+		// Open UDP socket to listen for Geneve Packets
+		conn, err := net.ListenPacket("udp", ":6081")
+		if err != nil {
```
Comment (plorenz) [inline router/xgress_geneve/listener.go:37]: Once you've got the `WithError` you don't need it in the message anymore.

`log.WithError(err).Error("failed to open geneve interface - udp")`

Comment (dariuszSki) [reply router/xgress_geneve/listener.go:37]: Good catch thanks!, updated it

File: router/xgress_geneve/listener.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+			log.WithError(err).Errorf("failed to open geneve interface - udp")
+			// error but return gracefully
+			return
```
Comment (plorenz) [inline router/xgress_geneve/listener.go:40]: Can remove the `else` from around the log statement

File: router/xgress_geneve/listener.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+			log.Infof("geneve interface started successfully - udp: %s", conn.LocalAddr().String())
+		}
+		// Close it when done
+		defer conn.Close()
+		// Open a raw socket to send Modified Packets to Networking Stack
+		fd, err := syscall.Socket(syscall.AF_INET, syscall.SOCK_RAW, syscall.IPPROTO_RAW)
+		if err != nil {
+			log.WithError(err).Errorf("failed to open geneve interface - fd")
+			// error but return gracefully
+			return
```
Comment (plorenz) [inline router/xgress_geneve/listener.go:51]: Can remove the `else` from around the log statement

Comment (dariuszSki) [reply router/xgress_geneve/listener.go:51]: good point, will eliminate the else

**Pull request merged**